File: Semestr III/Scala, Ocaml, Java/Paradygmaty programowania/Scala/lista12/Consumer.scala
class Consumer(name: String, buf: Buffer) extends Thread(name) {
  override def run() {
    var msg = ""
    do {
      msg = buf.take
    } while (msg != "Done")
  }
}

object prodCons {
  def main(args: Array[String]) {
    val buf = new Buffer
    new Producer("Producer", buf).start
    new Consumer("Consumer", buf).start
  }
}

File: Semestr III/Scala, Ocaml, Java/Paradygmaty programowania/Java/Lista11/Consumer.java
package Lista11;

public class Consumer extends Thread {
    final private Consume buf;

    public Consumer(String name, Consume buf) {
        super(name);
        this.buf = buf;
    }

    @Override
    public void run() {
        for (int i = 1; i <= 100; i++) {
            System.out.println(getName() + " consumed " + buf.take());
        }
    }
}

